File: providers/aws/src/main/java/cloudspec/aws/AWSClientsProvider.java
/*-
 * #%L
 * CloudSpec AWS Provider
 * %%
 * Copyright (C) 2020 Ezequiel Foncubierta
 * %%
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 * 
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 * 
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 * #L%
 */
package cloudspec.aws;

import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.ec2.Ec2Client;
import software.amazon.awssdk.services.iam.IamClient;
import software.amazon.awssdk.services.s3.S3Client;
import software.amazon.awssdk.services.sqs.SqsClient;

public class AWSClientsProvider implements IAWSClientsProvider {
    @Override
    public IamClient getIamClient() {
        return IamClient.builder().build();
    }

    @Override
    public Ec2Client getEc2Client() {
        return Ec2Client.create();
    }

    @Override
    public Ec2Client getEc2ClientForRegion(String region) {
        return Ec2Client.builder().region(Region.of(region)).build();
    }

    @Override
    public S3Client getS3Client() {
        return S3Client.create();
    }

    @Override
    public SqsClient getSqsClient() {
        return SqsClient.create();
    }

    @Override
    public SqsClient getSqsClientForRegion(String region) {
        return SqsClient.builder().region(Region.of(region)).build();
    }
}

File: providers/aws/src/main/kotlin/cloudspec/aws/AWSClientsProvider.kt
/*-
 * #%L
 * CloudSpec AWS Provider
 * %%
 * Copyright (C) 2020 Ezequiel Foncubierta
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *      http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */
package cloudspec.aws

import software.amazon.awssdk.core.retry.RetryMode
import software.amazon.awssdk.regions.Region
import software.amazon.awssdk.services.dynamodb.DynamoDbClient
import software.amazon.awssdk.services.ec2.Ec2Client
import software.amazon.awssdk.services.iam.IamClient
import software.amazon.awssdk.services.kms.KmsClient
import software.amazon.awssdk.services.s3.S3Client
import software.amazon.awssdk.services.sns.SnsClient
import software.amazon.awssdk.services.sqs.SqsClient

class AWSClientsProvider : IAWSClientsProvider {
    override val dynamoDbClient: DynamoDbClient
        get() = DynamoDbClient.builder()
            .overrideConfiguration { builder -> builder.retryPolicy(RetryMode.STANDARD) }
            .build()

    override fun dynamoDbClientForRegion(region: Region): DynamoDbClient {
        return DynamoDbClient.builder()
            .overrideConfiguration { builder -> builder.retryPolicy(RetryMode.STANDARD) }
            .region(region)
            .build()
    }

    override val ec2Client: Ec2Client
        get() = Ec2Client.builder()
            .overrideConfiguration { builder -> builder.retryPolicy(RetryMode.STANDARD) }
            .build()

    override fun ec2ClientForRegion(region: String): Ec2Client {
        return Ec2Client.builder()
            .overrideConfiguration { builder -> builder.retryPolicy(RetryMode.STANDARD) }
            .region(Region.of(region))
            .build()
    }

    override val iamClient: IamClient
        get() = IamClient.builder()
            .overrideConfiguration { builder -> builder.retryPolicy(RetryMode.STANDARD) }
            .build()

    override val kmsClient: KmsClient
        get() = KmsClient.builder()
            .overrideConfiguration { builder -> builder.retryPolicy(RetryMode.STANDARD) }
            .build()

    override fun kmsClientForRegion(region: Region): KmsClient {
        return KmsClient.builder()
            .overrideConfiguration { builder -> builder.retryPolicy(RetryMode.STANDARD) }
            .region(region)
            .build()
    }

    override val s3Client: S3Client
        get() = S3Client.builder()
            .overrideConfiguration { builder -> builder.retryPolicy(RetryMode.STANDARD) }
            .build()

    override val snsClient: SnsClient
        get() = SnsClient.builder()
            .overrideConfiguration { builder -> builder.retryPolicy(RetryMode.STANDARD) }
            .build()

    override fun snsClientForRegion(region: Region): SnsClient {
        return SnsClient.builder()
            .overrideConfiguration { builder -> builder.retryPolicy(RetryMode.STANDARD) }
            .region(region)
            .build()
    }

    override val sqsClient: SqsClient
        get() = SqsClient.builder()
            .overrideConfiguration { builder -> builder.retryPolicy(RetryMode.STANDARD) }
            .build()

    override fun sqsClientForRegion(region: Region): SqsClient {
        return SqsClient.builder()
            .overrideConfiguration { builder -> builder.retryPolicy(RetryMode.STANDARD) }
            .region(region)
            .build()
    }
}
